wiring/src/spark_wiring_system.cpp

```diff
@@ -41,6 +41,52 @@ void SystemClass::reset(uint32_t data)
     HAL_Core_System_Reset_Ex(RESET_REASON_USER, data, nullptr);
 }
 
+int SystemClass::sleep(const SystemSleepConfiguration& config) {
+    int ret;
+#ifndef SPARK_NO_CLOUD
+    bool cloudAutoConnect;
+#endif
+    bool wifiEnabled;
+    bool wifiConnect;
+
+    if (config.sleepMode() == SystemSleepMode::NONE || !config.halCoreSleepConfig()->wakeup_sources) {
+        return SYSTEM_ERROR_INVALID_ARGUMENT;
+    }
+
+    if ((config.sleepMode() == SystemSleepMode::STOP && !config.wakeupSourceFeatured(HAL_CORE_WAKEUP_SOURCE_TYPE_NETWORK)) ||
```

Comment:
This piece of code is just a copy from the `system_sleep.cpp`. In sleep 2.0, the `system_sleep.cpp` and `system_sleep.h` will be deleted. `spark_wiring_system.cpp` will call the core HAL APIs directly.

Reply:
Let's have a discussion about this tomorrow. I think we might have to still have a proxy system level sleep method (still delegating to HAL) that might do a couple of things:
1. Deal with system level stuff
2. Deal with system level sleep modes (if we are to introduce any)
3. Potentially generate system events after wakeup
etc